src/formatDate.js

```diff
@@ -48,9 +48,34 @@
  *
  * @returns {string}
  */
-
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateCopy = date.split(fromFormat[3]);
+
+  let newArray = Array.from(toFormat.slice(0, 3));
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    for (let y = 0; y < toFormat.length - 1; y++) {
+      if (fromFormat[i] === toFormat[y]) {
+        newArray[y] = dateCopy[i];
+      }
+
+      if (fromFormat[i] === 'YYYY' && toFormat[y] === 'YY') {
+        newArray[y] = dateCopy[i].slice(2, 4);
+      }
+
+      if (fromFormat[i] === 'YY' && toFormat[y] === 'YYYY') {
+        if (dateCopy[i] < 30) {
+        newArray[y] = `20${dateCopy[i]}`;
+        }
+
+        if (dateCopy[i] >= 30) {
+          newArray[y] = `19${dateCopy[i]}`;
+        }
+      }
+    }
+  }
```

Comment:
Let's improve performance. We can avoid using nested loops by adding splited date to object like here:
`{
  'DD': 19,
  'MM': 02
  ...
}`

then we shouldn't iterate by the array to get access to specific property

Reply:
Done.